### ISSUE TITLE
`duplicate-names-in-diff-context` lint complains about lambda objects inside atom

### ISSUE THREAD

**Issue by h1alexbel** [objectionary/lints#577]
This is the EO code with atoms:

```eo
# Foo.
[] > foo
  [] > bar ?
  [] > buz ?
```

However, `duplicate-names-in-diff-context` complains about `λ` inside the `bar` and `buz` atoms like this:

```text
Object λ has same name as object on line 3 (duplicate-names-in-diff-context (Single))
Object λ has same name as object on line 4 (duplicate-names-in-diff-context (Single))
```

We should not complain about lambda objects, if they are nested in the atoms.

**Comment by h1alexbel** [comment]
@yegor256 new bug

**Comment by Marat-Tim** [comment]
@h1alexbel Let me fix it

What other names are considered "reserved"?
I know about `@`, `λ`, and names generated using `>>`
Are there any others?

**Comment by h1alexbel** [comment]
@Marat-Tim Sure, go ahead! The attributes we need to cover are: `@` and `λ`. Auto named objects (`>>`) are generally unique, so there is no point in checking them

**Comment by VasilevNStas** [comment]
@yegor256 @volodya-lombrozo @maxonfjvipon, this issue is **already resolved on `master`**: the `duplicate-names-in-diff-context` lint has been removed in commit `f979fe0f` (refactor(#972)). The lambda (`λ`) false-positive you reported was one of the reasons the lint was removed; there is no corresponding XSL file in `src/main/resources` anymore.

I'd suggest closing this issue as resolved-by-removal (the lint no longer exists).

**Comment by volodya-lombrozo** [comment]
@h1alexbel We don't have `duplicate-names-in-diff-context` anymore. Closing thing one. Thank you for reporting!
